main.rb

```diff
@@ -0,0 +1,172 @@
+module Enumerable
+  def my_each
+    pos = 0
+    loop do
+      break if pos >= length
+
+      yield self[pos]
+      pos += 1
+    end
+  end
+
+  def my_each_with_index
+    pos = 0
+    loop do
+      break if pos >= length
+
+      yield self[pos], pos
+      pos += 1
+    end
+  end
+
+  def my_select
+    return_arr = []
+    my_each do |element|
+      return_arr.push(element) if yield element
+    end
+    return_arr
+  end
+
+  def my_all
+    return_val = true
+    my_each do |element|
+      return_val = yield element
+    end
+    return_val
+  end
+
+  def my_any
+    return_val = false
+    my_each do |element|
+      return_val = yield element
+    end
+    return_val
+  end
+
+  def my_none
+    return_val = true
+    my_each do |element|
+      return_val = !(yield element)
+    end
+    return_val
+  end
+
+  def my_count(arg = nil)
+    count_val = 0
+    my_each do |element|
+      count_val += 1 if (block_given? && yield(element)) || (element == arg)
+    end
+    count_val
+  end
```

Comment:
* [ ] `my_count` is identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_count returns the number of items in enum through enumeration
  ```ruby
    array = [1,2,5,4,7]
    p array.my_count == array.count # false
  ```